crates/conversions/src/felt252.rs

```diff
@@ -69,6 +83,47 @@ impl FromShortString<Felt252> for Felt252 {
     }
 }
 
+pub trait TryInferFormat: Sized {
+    /// Parses value from `hex string`, `dec string` and `quotted cairo shortstring`
+    fn infer_format_and_parse(value: &str) -> Result<Vec<Self>, ParseFeltError>;
+}
+
+impl<T> TryInferFormat for T
+where
+    T: From<Felt252>,
+{
+    fn infer_format_and_parse(mut value: &str) -> Result<Vec<Self>, ParseFeltError> {
+        let mut result = Err(ParseFeltError);
+
+        let is_short_string = value.starts_with('\'') && value.ends_with('\'');
+        let is_string = value.starts_with('"') && value.ends_with('"');
+
+        if is_short_string || is_string {
+            value = &value[1..value.len() - 1];
+        }
+
+        if is_short_string {
+            let value_escaped = value.replace("\\n", "\n").replace("\\'", "'");
+
+            result =
+                result.or_else(|_| Felt252::from_short_string(&value_escaped).map(|f| vec![f]));
+        }
+
+        if is_string {
+            //TODO
```

Comment:
let's improve on todo, preferably add issue number

Reply:
Missed it, outdated